baybe/constraints/discrete.py

```diff
@@ -287,10 +355,26 @@ def get_invalid(self, data: pd.DataFrame) -> pd.Index:  # noqa: D102
     DiscreteLinkedParametersConstraint,
     DiscreteSumConstraint,
     DiscreteProductConstraint,
+    DiscreteCardinalityConstraint,
     DiscretePermutationInvarianceConstraint,
     DiscreteDependenciesConstraint,
 )
 
+# Discrete constraints valid only for numeric parameters. There are three options:
+# Option A: A list containing such discrete constraints are maintained. It requires
+# no breaking change but requires some maintenance work, meaning whenever each new
+# discrete constraint (for numerical parameter only) must be added to the list.
+# Option B: Introduce an attribute, e.g. numerical_parameter_only(default = False).
+# It lead to breaking change: update of initialization of DiscreteSumConstraint and
+# DiscreteProductConstraint.
+# Option C: redesign of constraint module, differentiate between
+# numerical_parameter_only constraints and others.
+# Option A is chosen at the moment.
+DISCRETE_CONSTRAINTS_ONLY_FOR_NUMERIC_PARAMETER = [
```

Comment:
I don't see how this is any related to the PR?

Reply:
I think the discrete constraints (listed in `DISCRETE_CONSTRAINTS_ONLY_FOR_NUMERIC_PARAMETER`)  are only valid for the numerical discrete parameters. It is used for the validation purpose, see `validate_constraints` in `baybe/constraints/validation.py`. Shall I remove the validation in this PR?

Reply:
Make sense. I added `numerical_only` to `DiscreteConstraint` and the list `DISCRETE_CONSTRAINTS_ONLY_FOR_NUMERIC_PARAMETER` is not needed now.